src/Workspaces/Core/Portable/Execution/PinnedRemotableDataScope.cs

```diff
@@ -62,27 +63,36 @@ internal sealed class PinnedRemotableDataScope : IDisposable
 
         public readonly PinnedSolutionInfo SolutionInfo;
 
-        public PinnedRemotableDataScope(
+        private PinnedRemotableDataScope(
             AssetStorages storages,
             AssetStorages.Storage storage,
-            Checksum solutionChecksum)
+            PinnedSolutionInfo solutionInfo)
         {
-            Contract.ThrowIfNull(solutionChecksum);
-
             _storages = storages;
             _storage = storage;
+            SolutionInfo = solutionInfo;
+        }
 
-            SolutionInfo = new PinnedSolutionInfo(
+        public Workspace Workspace => _storage.SolutionState.Workspace;
```

Comment:
Nit: I'd place these after the factory. It helps to have factories close to constructors.

Reply:
I kept the relative ordering in this file for now (constructors, then properties, then methods).